Mas ghafix

martinsumner opened the pull request:
It appears the there is an issue with support for `chmod` within our Github Actions setup - in that it seems to return status of 0 to any chmod request, but doesn't change permissions as expected.

To progress, tests can now be marked with `ifndef(GITHUBEXCLUDE).` and they will not be run on gtihub actions, but will continue to be run within local `rebar3 eunit` test runs.

martinsumner committed:
Update erlang.yml

martinsumner committed:
Update erlang.yml

martinsumner committed:
Update erlang.yml

martinsumner committed:
Update riak_kv_vnode_status_mgr.erl

martinsumner committed:
Update riak_kv_vnode_status_mgr.erl

martinsumner committed:
Update riak_kv_vnode_status_mgr.erl

martinsumner committed:
Update riak_kv_vnode_status_mgr.erl

martinsumner committed:
Update riak_kv_vnode_status_mgr.erl

martinsumner committed:
Revert "Update riak_kv_vnode_status_mgr.erl"

This reverts commit 114c30c49543021a5c01c19359ce1ec6c82f83b4.

martinsumner committed:
Simplest form of test

martinsumner committed:
Update riak_kv_vnode_status_mgr.erl

martinsumner committed:
Be strict about checking for write failure

martinsumner committed:
Test chmod existence (man)

martinsumner committed:
bash shell (explicit)

martinsumner committed:
Update erlang.yml

martinsumner committed:
Update riak_kv_vnode_status_mgr.erl

martinsumner committed:
Exclude non GHA friendly test

martinsumner committed:
Attempt to exclude tests in GHA

martinsumner committed:
Use profile to exclude tests on GHA

martinsumner committed:
Indentation correction

martinsumner committed:
Merge branch 'develop-3.0' into mas-ghafix

martinsumner committed:
don't trigger on every push

ThomasArts commented on rebar.config:
```diff
@@ -43,7 +43,10 @@
                     {eunit_formatters, ".*", {git, "git://github.com/seancribbs/eunit_formatters", {tag, "v0.5.0"}}}]}]},
-    {eqc, [{deps, [meck]}]}
+    {eqc, [{deps, [meck]}]},
+    {gha, [{deps, [meck,
```
I think you should not need the meck stuff. Unit will by default run the test profile and hence will use all in test as well as gha profile in above definition of "Run unit".

martinsumner replied:
Removed deps from the gha profile

ThomasArts commented on rebar.config:
```diff
@@ -43,7 +43,10 @@
                     {eunit_formatters, ".*", {git, "git://github.com/seancribbs/eunit_formatters", {tag, "v0.5.0"}}}]}]},
-    {eqc, [{deps, [meck]}]}
+    {eqc, [{deps, [meck]}]},
+    {gha, [{deps, [meck,
```
```suggestion
    {gha, [{deps, [
```

ThomasArts commented on rebar.config:
```diff
@@ -43,7 +43,10 @@
-    {eqc, [{deps, [meck]}]}
+    {eqc, [{deps, [meck]}]},
+    {gha, [{deps, [meck,
+                    {eunit_formatters, ".*", {git, "git://github.com/seancribbs/eunit_formatters", {tag, "v0.5.0"}}}]},
```
```suggestion
                    
```

ThomasArts commented on .github/workflows/erlang.yml:
```diff
@@ -1,10 +1,7 @@
-    branches: [ develop-3.0 ]
-  pull_request:
-    branches: [ develop-3.0 ]
+  pull_request
```
As comment: I guess that this works, since only branches that have this .yml file in it do act on the pull requests and branches that are based on old rebar version simply don't have this file.

martinsumner replied:
I'm not sure what the right configuration is, but I've reverted to make consistent with the other repos for now.

ThomasArts commented on .github/workflows/erlang.yml:
```diff
@@ -16,8 +13,10 @@ jobs:
+      - name: Compile
+        run: ./rebar3 compile
+      - name: Run xref and dialyzer
+        run: rebar3 do xref, dialyzer
```
```suggestion
        run: ./rebar3 do xref, dialyzer
```

martinsumner replied:
Done

ThomasArts commented on .github/workflows/erlang.yml:
```diff
@@ -16,8 +13,10 @@ jobs:
+      - name: Run xref and dialyzer
+        run: rebar3 do xref, dialyzer
+      - name: Run eunit
+        run: rebar3 as gha do eunit
```
```suggestion
        run: ./rebar3 as gha do eunit
```

I assume you want to run the rebar version supplied not to get surprised by incompatibilities in rebar?

martinsumner replied:
Done

ThomasArts commented:
I advice to squash the commits before merging

martinsumner committed:
max integer()/undefined -> undefined

As a safety check the definition of CSP checks it as at least 1 to avoid badarith when divided.  However, when called from eqc tests, this property will be undefined (should be OK when started via Riak), and max(undefined, 1) is undefined.  So this needs a default to work safely from tests

martinsumner committed:
Recommended fixes

Following review from Thomas

martinsumner committed:
Revert to develop-3.0 restriction

Not sure what the best setting is, but for now reverted to make consistent with other GHA settings in the riak environment
